packages/cli/src/commands/pg/credentials/rotate.ts

```diff
@@ -0,0 +1,92 @@
+import color from '@heroku-cli/color'
+import {Command, flags} from '@heroku-cli/command'
+import {APIClient} from '@heroku-cli/command/lib/api-client'
+import type {AddOnAttachment} from '@heroku-cli/schema'
+import {Args, ux} from '@oclif/core'
+import confirmApp from '../../../lib/apps/confirm-app'
+import {attachment} from '../../../lib/pg/fetcher'
+import host from '../../../lib/pg/host'
+import {legacyEssentialPlan} from '../../../lib/pg/util'
+
+export default class Rotate extends Command {
+  static topic = 'pg'
+  static description = 'rotate the database credentials'
+  static flags = {
+    name: flags.string({
+      char: 'n',
+      description: 'which credential to rotate (default credentials if not specified and --all is not used)',
+    }),
+    all: flags.boolean({description: 'rotate all credentials'}),
+    confirm: flags.string({char: 'c'}),
+    force: flags.boolean({description: 'forces rotating the targeted credentials'}),
+    app: flags.app({required: true}),
+  }
+
+  static args = {
+    database: Args.string(),
+  }
+
+  public async run(): Promise<void> {
+    const {flags, args} = await this.parse(Rotate)
+    const {app, all, confirm, name, force} = flags
+    const {addon: db} = await attachment(this.heroku, app, args.database)
+    const warnings: string[] = []
+    const cred = name || 'default'
+    if (all && name !== undefined) {
+      throw new Error('cannot pass both --all and --name')
```

Comment:
we can prevent this with the `exclusive` oclif helper. that is,

```javascript
  all: flags.boolean({exclusive: ['name']})
```

Reply:
+1